jux/state/state.py

```diff
@@ -1588,7 +1588,15 @@ def _cache_water_info(self, factory_actions: Array) -> Tuple[Array, Array]:
         # -9 for the factory occupied cells
         grow_lichen_size = cmp_cnt[self.factories.pos.x, self.factories.pos.y] - 9  # int[2, F]
 
-        return color, grow_lichen_size
+        # compute number of connected lichen tiles
+        def loop_fn(_, x):
+            y_0 = jnp.where(x[0] != imax(x.dtype), (connected_lichen == x[0]).sum(), 0)
+            y_1 = jnp.where(x[1] != imax(x.dtype), (connected_lichen == x[1]).sum(), 0)
+            return None, jnp.array([y_0, y_1])
+        _, connected_lichen_size = jax.lax.scan(loop_fn, None, self.factories.num_id.T)
```

Comment:
is there faster way to compute this? probably not a compute issue anyway since theres max 12 factories

Reply:
I will take care of it.